bin/docker-build-proxy

```diff
@@ -13,5 +13,15 @@ rootdir="$( cd $bindir/.. && pwd )"
 . $bindir/_docker.sh
 . $bindir/_tag.sh
 
-docker_build proxy "$(head_root_tag)" $rootdir/proxy/Dockerfile \
-    --build-arg="PROXY_UNOPTIMIZED=${PROXY_UNOPTIMIZED:-}"
+if [ -z "${PROXY_BUILD_CACHE_IMAGE:-}" ]; then
+    # If a cache is not specified use the default docker cache.
+    docker_build proxy "$(head_root_tag)" $rootdir/proxy/Dockerfile \
+        --build-arg="PROXY_UNOPTIMIZED=${PROXY_UNOPTIMIZED:-}"
+
+else
+    # Otherwise, if a cache image is specified, use it.
+    docker_build proxy "$(head_root_tag)" $rootdir/proxy/Dockerfile \
+        --build-arg="PROXY_UNOPTIMIZED=${PROXY_UNOPTIMIZED:-}" \
+        --cache-from="$PROXY_BUILD_CACHE_IMAGE" \
+        --target runtime
```

Comment:
NIT: Isn't `--target runtime` the default? It is a bit confusing that we use `--target runtime` in the `else` branch but not the `if` branch.